CRM/Upgrade/Incremental/General.php

```diff
@@ -66,12 +71,26 @@ class CRM_Upgrade_Incremental_General {
    */
   public static function setPreUpgradeMessage(&$preUpgradeMessage, $currentVer, $latestVer) {
     if (version_compare(phpversion(), self::MIN_RECOMMENDED_PHP_VER) < 0) {
-      $preUpgradeMessage .= '<p>' .
-        ts('This webserver is running an outdated version of PHP (%1). It is strongly recommended to upgrade to PHP %2 or later, as older versions can present a security risk.', array(
-          1 => phpversion(),
-          2 => self::MIN_RECOMMENDED_PHP_VER,
-        )) .
-        '</p>';
+      $preUpgradeMessage .= '<p>';
+      // CRM-20941 PHP 5.3 end date of End of 2017, PHP 5.4 End date End of Feb 2019 Recommend anyone on PHP 5.5 to move up to 5.6 or later e.g. 7.0
+      if (version_compare(phpversion(), self::PREVIOUS_MIN_RECOMMENDED_PHP_VER) >= 0) {
+        $preUpgradeMessage .= ts('You may proceed with the upgrade and CiviCRM %1 will continue working normally, but future releases will require PHP %2 or above. We recommend you use the most recent php version you can', array(
+           1 => $currentVer,
+           2 => self::MIN_RECOMMENDED_PHP_VER,
+        ));
+      }
+      elseif (version_compare(phpversion(), 5.5) < 0) {
+        $date = 'February 2018';
+        if (version_compare(phpversion(), 5.4) < 0) {
+          $date = 'the end of 2017';
+        }
+        $preUpgradeMessage .= ts('You may proceed with the upgrade and CiviCRM %1 will continue working normally, but PHP %2 will not work in releases published after %3. We recommend you use the most recent php version you can', array(
+          1 => $currentVer,
+          2 => phpversion(),
+          3 => $date,
```

Comment:
@seamuslee001, please either use a localized date as an argument to the string (using `CRM_Utils_Date::formatDate()`) or create two separate strings (I prefer the former, even if it sounds more machine-like, admins will read this only once). Translators will not understand why your sentence does not end with a period, or how to conjugate "the end of 2017", if it had a `ts`.